app/src/components/common/text/section_title/index.tsx

```diff
@@ -46,7 +46,7 @@ export const SectionTitleWrapper: React.FC<Props> = (props: Props) => {
   return (
     <Wrapper {...restProps}>
       {enableGoBack && (
-        <ButtonCircle onClick={() => props.history.goBack()}>
+        <ButtonCircle onClick={() => (props.history.length > 2 ? props.history.goBack() : props.history.replace('/'))}>
```

Comment:
This fails if there is any history prior to visiting the create market page. For example, visit a market -> go to buy -> go to create market -> taken to market details page -> back button goes to create market page

Reply:
Although not perfect, considering the complex routing for filters, this may be the best fix at the moment.

Reply:
The idea is to replace the history item related to market creation with the detail (replace instead of push), that piece of code is to have a working back button when you entered directly to the market.
I'm not sure if react history gives us enough flexibility to do much more without a custom solution.

Reply:
Maybe reverting changes to use a goBack string could work for more complex scenarios, but in that case, it should fallback too to a default route too. Let me know if more changes are needed.